mmeval/metrics/rouge.py

```diff
@@ -0,0 +1,307 @@
+        return ngrams
+
+    pred_ngarms = _create_ngrams(pred, n_gram)
+    reference_ngarms = _create_ngrams(reference, n_gram)
+    pred_len = sum(pred_ngarms.values())
+    reference_len = sum(reference_ngarms.values())
+    if pred_len == 0 or reference_len == 0:
+        return dict(precision=0., recall=0., fmeasure=0.)
+
+    # Take the intersection of n_gram of prediction and reference.
+    hits = sum(
+        min(pred_ngarms[w], reference_ngarms[w]) for w in set(pred_ngarms))
+    return _get_results(hits, pred_len, reference_len)
+
+
+class ROUGE(BaseMetric):
+    """Calculate Rouge Score used for automatic summarization.
+
+    This metric proposed in `ROUGE: A Package for Automatic Evaluation
+    of Summaries <https://www.aclweb.org/anthology/W04-1013.pdf>`_ are
+    common evaluation indicators in the fields of machine translation,
+    automatic summarization, question and answer generation, etc.
+
+    Args:
+         rouge_keys (List or Tuple or int or str):
+            A list of rouge types to calculate.
+            Keys that are allowed are ``L``, and ``1`` through ``9``.
+            Defaults to ``(1, 2, 'L')``.
+        use_stemmer (bool): Use Porter stemmer to strip word
+            suffixes to improve matching. Defaults to False.
+        normalizer (Callable, optional): A user's own normalizer function.
+            If this is ``None``, replacing any non-alpha-numeric characters
+            with spaces is default. Defaults to None.
+        tokenizer (Callable or str, optional): A user's own tokenizer function.
+            Defaults to None.
+        accumulate (str): Useful in case of multi-reference rouge score.
+            ``avg`` takes the average of all references with respect to predictions  # noqa: E501
+            ``best`` takes the best fmeasure score obtained between prediction
+            and multiple corresponding references.
+            Defaults to ``best``.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+
+        >>> from mmeval import ROUGE
+        >>> predictions = ['the cat is on the mat']
+        >>> references = [['a cat is on the mat']]
+        >>> metric = ROUGE(rouge_keys='L')
+        >>> metric.add(predictions, references)
+        >>> results = metric.compute_metric()
+        {'rougeL_fmeasure': 0.8333333,
+         'rougeL_precision': 0.8333333,
+         'rougeL_recall': 0.8333333}
+    """
+
+    def __init__(
+        self,
+        rouge_keys: Union[List, Tuple, int, str] = (1, 2, 'L'),
+        use_stemmer: bool = False,
+        normalizer: Optional[Callable] = None,
+        tokenizer: Union[Callable, str, None] = None,
+        accumulate: str = 'best',
+        **kwargs: Any,
+    ):
+        super().__init__(**kwargs)
+        assert accumulate in ['best', 'avg'], \
+            f'Wrong accumulate {accumulate}. Supported accumulate are "best" and "avg"'  # noqa: E501
+        self.accumulate = accumulate
```

Comment:
It is better to process parameters in the order of passing parameters.

Suggest move L150-L152 to the end of \_\_init\_\_ function~